components/camel-resteasy/src/main/java/org/apache/camel/component/resteasy/DefaultResteasyHttpBinding.java

```diff
@@ -0,0 +1,219 @@
+    }
+
+    @Override
+    public Response populateResteasyRequestFromExchangeAndExecute(String uri, Exchange exchange, Map<String, String> parameters) {
+        Client client = ClientBuilder.newBuilder().build();
+        String body = exchange.getIn().getBody(String.class);
+
+        LOG.debug("Body in producer: " + body);
+
+        String mediaType = exchange.getIn().getHeader(Exchange.CONTENT_TYPE, String.class);
+
+        WebTarget target = client.target(uri);
+
+        LOG.debug("Populate Resteasy request from exchange body: {} using media type {}", body, mediaType);
+
+        Invocation.Builder builder;
+        if (mediaType != null) {
+            builder = target.request(mediaType);
+        } else {
+            builder = target.request();
+        }
+
+
+        for (Map.Entry<String, Object> entry : exchange.getIn().getHeaders().entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+            if (headerFilterStrategy != null
+                    && !headerFilterStrategy.applyFilterToCamelHeaders(key, value, exchange)) {
+                builder.header(key, value);
+                LOG.debug("Populate Resteasy request from exchange header: {} value: {}", key, value);
+            }
+        }
+
+        if (parameters.get("username") != null && parameters.get("password") != null) {
+            target.register(new BasicAuthentication(parameters.get("username"), parameters.get("password")));
+        }
+        LOG.debug("Basic authentication was applied");
+        String method = parameters.get("method");
+
+        if (method.equals("GET")) {
+            return builder.get();
+        }
+        if (method.equals("POST")) {
+            return  builder.post(Entity.entity(body, mediaType));
+        }
+        if (method.equals("PUT")) {
+            return  builder.put(Entity.entity(body, mediaType));
+        }
+        if (method.equals("DELETE")) {
+            return  builder.delete();
+        }
+        if (method.equals("OPTIONS")) {
+            return  builder.options();
+        }
+        if (method.equals("TRACE")) {
+            return  builder.trace();
+        }
+        if (method.equals("HEAD")) {
+            return  builder.head();
+        }
+
+        // maybe throw exception because not method was correct
+        throw new IllegalArgumentException("Method '" + method + "' is not supported method");
+    }
+
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    @Override
+    public void populateProxyResteasyRequestAndExecute(String uri, Exchange exchange, Map<String, String> parameters) {
+        Client client = ClientBuilder.newBuilder().build();
+
+        WebTarget target = client.target(uri);
+
+        if (parameters.get("username") != null && parameters.get("password") != null) {
+            target.register(new BasicAuthentication(parameters.get("username"), parameters.get("password")));
```

Comment:
Hmm I would rather have no surprise, and that we have a autoMethod=Basic you configure on the endpoint or component.

Reply:
not sure if i got this one right? this might be something i am missing. can you please give a little bit more details?

Reply:
You just assume username/password requires basic auth. But they may just be query parameters for something else. Instead we should have an option to turn on authentication first. And so only do this if that option is turned on.

Reply:
done